consensus/state.go

```diff
@@ -909,6 +909,9 @@ func (cs *ConsensusState) defaultDecideProposal(height int64, round int) {
 		}
 	}
 
+	// Flush the WAL. Otherwise, we may not recompute the same proposal to sign, and the privValidator will refuse to sign anything.
```

Comment:
should this happen in here or right before we call `decideProposal` ? If we keep it as is, it means any other implementations of the `decideProposal` functionality besides this default need to also remember to flush the WAL first. Might not actually be an issue though.